pkg/controller/nodenetworkconfigurationpolicy/conditions/manager.go

```diff
@@ -49,19 +51,22 @@ func (m *Manager) NotifySuccess() {
 }
 
 func (m *Manager) updateEnactmentCondition(
-	conditionsSetter func(*nmstatev1alpha1.EnactmentList, string, string),
+	conditionsSetter func(*nmstatev1alpha1.NodeNetworkConfigurationEnactment, string),
```

Comment:
Could we give the callback only list of Conditions as a parameter? Make sure they don't touch what they should not and the interface is slimmer?

Reply:
We may want to implement some counter or the like or even if we go for phase passing the whole enactment is a good thing also, we are not a library the "they" is us.

Reply:
And we want to protect ourselves.

When reading a code, having smaller arguments is better since you know what can be changed or consider inside the function without looking into it and reading it.

If we will be refactoring the code, it is easier to move such a function around and make it generic if needed.

Smaller signature makes it easier to provide data when testing it. When we change the big structure (Enactment) we may need to change tests of this small function even though it would no be directly affected.

The function is mean to update a condition. Would you please give an example of when we may use the whole function?

Reply:
I was thinking about the missing enactment state "Counter of reconfigurations" but since this is different from conditions, we are ok I am going to pass a ConditionList